### DISCUSSION TITLE
How not to invert `iframe` on Yandex.Practicum?

### DISCUSSION THREAD

**Discussion by rakleed** [darkreader/darkreader#9997] [Q&A]
Hello,

There is such a platform [Yandex.Practicum](https://practicum.yandex.ru) (or [Practicum](https://practicum.com) in English) for education. And it has web development courses. And on the site, there is a simulator in which you need to enter the correct code, and then it checks the result of the execution partly by tests, and partly on the client side. But since Dark Reader can change styles even for `iframe`, then in this case it becomes a problem, because when Dark Reader is enabled, some tasks give an error when checking.

Here is an archive with a copy of the page: [ypracticum.zip](https://github.com/darkreader/darkreader/files/9701113/ypracticum.zip)

I tried adding rules `.trainer-result__content { filter: none !important; }` and `.web-result-view__container { filter: none !important; }` in Dark Reader's Dev tools and also in Stylus extension, but it didn't work for either Dynamic or Filter/Filter+. I took this rule from https://github.com/darkreader/darkreader/discussions/9897 but apparently, it doesn't work for `iframe`.

I also tried to add the `NO INVERT .trainer-result__content` and `NO INVERT .web-result-view__container` rules to the Dark Reader's Dev tools for this site with the Filter/Filter+ mode enabled, but that didn’t help either.

Is it possible somehow not to invert `iframe` with `.trainer-result__content` class?

**Comment by Gusted** [comment]
Hi @rakleed!

I wasn't able to find the mentioned iframe with class `trainer-result__content` in the zip files.

Are you sure that the `<iframe` is being inverted? Or is darkreader simply modifying its contents? Does the `<iframe>` have a src attribute?

**Reply by rakleed** [reply]
When you unpack the archive and open `Яндекс Практикум.html` in Chrome, can't you find an `iframe` with this class in DevTools?

And yes, rather the content itself in the `iframe` is inverted.

`src` attribute value: `./Яндекс Практикум_files/frame.html`

<details><summary>DevTools screenshot</summary>

![Screenshot 2022-10-08 at 02 54 39](https://user-images.githubusercontent.com/19418601/194676473-88eea49a-d5ef-43f7-8ac6-d1331b475a6c.png)

</details>

**Reply by Gusted** [reply]
> And yes, rather the content itself in the `iframe` is inverted.

Unfortunately not, looking now again (and having the site actually be in the foreground) it looks like I wasn't seeing it due to:
![image](https://user-images.githubusercontent.com/25481501/194747127-57112205-bbe3-460b-adf3-4ce6d861877a.png)

**Comment by rakleed** [comment]
It looks like you're running a server, but why? When I simply opening `html` file, all other scripts open correctly. Screen recording: https://share.cleanshot.com/NNebyj

**Reply by Gusted** [reply]
> It looks like you're running a server, but why?

I'd prefer avoiding opening from local file system, Dark Reader doesn't work that great when dealing with local files. 

So seems like via that way it does work, but now the question? Where do I need to observe the filter being applied? As in my eyes these seems to work as intended.

**Reply by rakleed** [reply]
Dark Reader changes styles inside `iframe` with class `.trainer-result__content`. Is it possible to make the content inside this `iframe` not be modified by Dark Reader?

**Reply by Gusted** [reply]
Yeah that would be possible, you would need to insert the absolute path of that iframe src (so in this case, something along the lines of `http://site.com/aaaa/bbb/frame.html`) in the sitelist, Dark Reader shouldn't modify the frame then.

**Reply by rakleed** [reply]
The address of the main `iframe` is `https://feynman.praktikum.yandex.net:443/staticfiles/frame.html?v=0.185.0&language=en`. I tried adding it to the list of sites where Dark Reader should be disabled in all of these options, but that didn't help:

- `https://feynman.praktikum.yandex.net:443/staticfiles/frame.html?v=0.185.0&language=en`
- `https://feynman.praktikum.yandex.net:443/staticfiles/frame.html`
- `feynman.praktikum.yandex.net:443/staticfiles/frame.html?v=0.185.0&language=en`
- `feynman.praktikum.yandex.net:443/staticfiles/frame.html`

Then I also tried adding `iframe` that is embedded in it, but that didn't work either:

- `https://feynman.praktikum.yandex.net/staticfiles/blank.html`
- `feynman.praktikum.yandex.net/staticfiles/blank.html`

<details><summary>Screenshot</summary>

![Screenshot 2022-10-18 at 04 30 06](https://user-images.githubusercontent.com/19418601/196314586-c1b3630e-7112-4e8b-b39b-e0700d13d1ca.png)

</details>

Am I doing something wrong?

**Reply by Gusted** [reply]
That's weird, I thought the code covered to check if the frame's URL is in the sitelist, but apparently that check never existed. I will make a PR soon to add that check, no workaround for now...

**Comment by PilaScat** [comment]
still problematic, it worked fine since some months ago
